### DISCUSSION TITLE
Freeplane 1.11.3_01 preview and my new ChatGPT Communicator Add-On

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/freeplane#1156] [Announcements]
I extended scripting API to make it better work with ChatGPT.
I also developed a script and addon based on https://github.com/barrymac/freeplane_openai_addon and going far beyond it.
The script and the add-on are currently identical in functionality. You can find them at https://sourceforge.net/projects/freeplane/files/freeplane%20preview/ and at https://github.com/dpolivaev/freeplane_openai_addon.

The add-on works only with Freeplane 1.11.3.

Using the new add-on you can experiment with different prompts. They consist of two parts, the system message and the user message.

You prepare

<img width="745" alt="ChatGptCommunicator" src="https://github.com/freeplane/freeplane/assets/1558052/5c9a73f7-c024-4a68-beb4-261b7462dd93">

-------------------------
Depending on the context user message is generated and send to ChatGPT together with the selected system message

<img width="1207" alt="ChatGptQuestion" src="https://github.com/freeplane/freeplane/assets/1558052/657f0c42-035b-4616-84b5-fcc342f5b6a5">

-------------------------
And you get a rich answer as a mind map
<img width="1109" alt="ChatGptAnswer" src="https://github.com/freeplane/freeplane/assets/1558052/6dd37919-a063-4414-87f4-7e3650474da1">

I have further ideas. Also the documentation is still work in progress. But I think that you can now check it out.

@barrymac

**Comment by dpolivaev** [comment]
And if the question is asked at a non-root node, the contents of all ancestor nodes and all siblings nodes are added to the user message as "known facts".

**Comment by dpolivaev** [comment]
I have improved the add-on, the last version is v0.3.1. It is available at https://sourceforge.net/projects/freeplane/files/freeplane%20preview/

**Reply by spiritusanimalis** [reply]
Hi dpolivaev,
I'm running your Preview version of the upcoming latest release of Freeplane. I tried to configure the latest version of the ChatGPT add-on, by navigating to Add-ons, and trying to locate its Configure button, but I cannot find it. 

I've attached a screenshot. 
![Freeplane ChatGPT AddOn in Add-ons Screenshot 2023-05-22 142343 ](https://github.com/freeplane/freeplane/assets/102841572/3efeeb58-c392-48db-97ea-3069251b2191)

Am I missing something?

Regards

**Reply by dpolivaev** [reply]
Run the action from the Open AI menu, it should display a dialog with all configuration options and the "Ask GPT" button.

**Reply by spiritusanimalis** [reply]
Update
I ran the add on, pasting in my API key in the question pane: 

![Freeplane ChatGPT Screenshot 2023-05-22 144322](https://github.com/freeplane/freeplane/assets/102841572/131ae0c1-36d4-4fe4-be57-8def2506a533)

And got this...

![Freeplane ChatGPT Screenshot 2023-05-22 144702](https://github.com/freeplane/freeplane/assets/102841572/29205742-44b5-4dad-a619-da47c3b5aedb)

Perhaps this is because I'm using the Free Research Preview. [ChatGPT May 12 Version] rather than the paid version. C'est la guerre… LOL.

I will try again later, but this shows that everything works fine.

Thank you for your work

**Reply by dpolivaev** [reply]
The free preview is only available for the first 3 months after registration and only for a limited number of requests, later they want to pay with your credit card. But it is very very cheap.

**Reply by spiritusanimalis** [reply]
I've got the reminder to upgrade... we shall see.

**Comment by Illegit** [comment]
Is it, this addon, linkable to Bing using ChatGPT AI ?

**Reply by dpolivaev** [reply]
No

**Comment by Illegit** [comment]
I mean ChatGPT has been, at least partially, in use by Bing for some weeks. Is possible that the addon can choose to connect to Bing currently in-use AI,  instead of to ChatGPT?

**Reply by dpolivaev** [reply]
No

**Comment by barrymac** [comment]
Nice @dpolivaev ! Great to see this brought into freeplane, a software I've loved for many many years since back in the freemind days! 

One of the first things I tried with ChatGPT was to see if it was any good at producing freeplane files, which at the time it failed miserably at. :-) 

I would love to know any things about the new dependency on the latest freeplane version that would be good to know about regarding add-on development or this effort specifically? 

This was my first attempt at a plugin, as evidenced by some basics and UX conveniences that were still lacking that you've brought in. Also I was travelling without a laptop for a few weeks so was not able to continue or respond to pull requests. 

I was also aspiring to extend the "prompt engineering" support as you have done! 

It's interesting as well to see a slightly different take on some ideas.

**Comment by wordmuse** [comment]
This could be so cool! I'm paying for a ChatGPT account. I want to try this!

I tried installing the add-on, but go this message:
<img width="710" alt="show-addons-dialog" src="https://github.com/freeplane/freeplane/assets/8197883/d3db0080-3070-405b-b875-a18790a69919">

As you can see it's in Freeplane's Addons folder.  Is there something I need to do before trying to install this addon?

Thanks.
~ Bal
